src/v3/recordingComputeUtils.ts

```diff
@@ -89,50 +205,74 @@ export function getComputedSpans<
   for (const [name, computedSpanDefinition] of Object.entries(
     context.definition.computedSpanDefinitions,
   )) {
-    const { startSpan: startSpanMatcher, endSpan } = computedSpanDefinition
+    // Create matchers from the span definitions
+    const startSpanMatcher = createMatcher<
+      SelectedRelationNameT,
+      RelationSchemasT,
+      VariantsT
+    >(computedSpanDefinition.startSpan)
+
+    const endSpanMatcher = createMatcher<
+      SelectedRelationNameT,
+      RelationSchemasT,
+      VariantsT
+    >(computedSpanDefinition.endSpan)
+
+    // Find matching start entry
+    let matchingStartEntry:
+      | SpanAndAnnotation<RelationSchemasT>
+      | 'operation-start'
+      | undefined =
+      startSpanMatcher === 'operation-start' ? 'operation-start' : undefined
+
+    if (typeof startSpanMatcher === 'function') {
+      matchingStartEntry = findMatchingSpan(
+        startSpanMatcher,
+        recordedItemsArray,
+        context,
+      )
+    }
 
-    const matchingStartEntry =
-      typeof startSpanMatcher === 'function'
-        ? recordedItemsArray.find((spanAndAnnotation) =>
-            startSpanMatcher(spanAndAnnotation, context),
-          )
-        : startSpanMatcher
+    // Find matching end entry
+    let matchingEndEntry: SpanAndAnnotation<RelationSchemasT> | undefined
+
+    if (typeof endSpanMatcher === 'function') {
+      matchingEndEntry = findMatchingSpan(
+        endSpanMatcher,
+        recordedItemsArray,
+        context,
+      )
+    } else if (endSpanMatcher === 'operation-end') {
+      matchingEndEntry = findMatchingSpan(
+        markedComplete,
+        recordedItemsArray,
+        context,
+      )
+    } else if (endSpanMatcher === 'interactive') {
+      matchingEndEntry = findMatchingSpan(
+        markedInteractive,
+        recordedItemsArray,
+        context,
+      )
```

Comment:
'operation-end' and 'interactive' should iterate from the end (matchingIndex === -1) for performance.
Even better, we can just pass these in from `createTraceRecording` as arguments, there's both `completeSpanAndAnnotation` and `cpuIdleSpanAndAnnotation` passed in and we don't need to iterate/search at all.